accounts/management/commands/monitor_async_deleted_users.py

```diff
@@ -0,0 +1,127 @@
+#
+# Freesound is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Freesound is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     See AUTHORS file.
+#
+
+import logging
+
+import requests
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand
+
+from accounts.admin import DELETE_SPAMMER_USER_ACTION_NAME, FULL_DELETE_USER_ACTION_NAME, \
+    DELETE_USER_DELETE_SOUNDS_ACTION_NAME, DELETE_USER_KEEP_SOUNDS_ACTION_NAME
+from accounts.models import Profile
+
+logger = logging.getLogger('console')
+
+
+class Command(BaseCommand):
+    help = 'Gets a list of users that should have been asynchronously deleted recently and makes sure that these were' \
+           'indeed deleted. The list of users is taken from Graylog "Delete user request" log messages.'
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            '-n', '--n_days',
+            action='store',
+            dest='days',
+            default=15,
+            help='Number of days to search back for deletion requests')
+
+    def handle(self, *args, **options):
+
+        def get_messages_for_params(params):
+            auth = (settings.GRAYLOG_USERNAME, settings.GRAYLOG_PASSWORD)
+            r = requests.get(settings.GRAYLOG_DOMAIN + '/graylog/api/search/universal/relative/',
+                             auth=auth, params=params, headers={'Accept': 'application/json'})
+            response = r.json()
+            if 'total_results' not in response:
+                return -1, []
+            return response['total_results'], response['messages']
+
+        messages = []
+        results_remaining = True
+        messages_per_page = 150
+        params = {
+            'query': '"Requested async deletion of user"',
+            'range': 60 * 60 * 24 * int(options['days']),
+            'fields': 'message',
+            'limit': messages_per_page,
+            'offset': 0
+        }
+
+        while results_remaining:
+            page = (params['offset'] / messages_per_page) + 1
+            logger.info('Getting user delete requests info (page {0})'.format(page))
+
+            try:
+                total_results, new_messages = get_messages_for_params(params)
+            except requests.HTTPError as e:
+                logger.error('Can\'t access Graylog server. Error: ' + str(e))
```

Comment:
Prefer to use `"` instead of escaping quotes inside the string. If I need both `"` and `'` I will triple-quote the string

Reply:
Interesting bug here which I found recently in AcousticBrainz. If the message in `e` contains a unicode string, this will fail to render. Instead, use `u'got an error: {}'.format(e)`

Reply:
Addressed 671fbe4de71dc952484515ca669bbfd7878d1b25